infrastructure/network/connmanager/connmanager.go

```diff
@@ -147,13 +147,14 @@ func (c *ConnectionManager) IsBanned(netConnection *netadapter.NetConnection) (b
 func (c *ConnectionManager) waitTillNextIteration() {
 	select {
 	case <-c.resetLoopChan:
-		c.loopTicker.Stop()
-		c.loopTicker = time.NewTicker(connectionsLoopInterval)
+		c.loopTicker.Reset(connectionsLoopInterval)
 	case <-c.loopTicker.C:
 	}
 }
 
 func (c *ConnectionManager) connectionExists(addressString string) bool {
+	c.connectionRequestsLock.Lock()
```

Comment:
I think it's better to just add this in the exported function that calls it (to avoid unexpected deadlocks)

Reply:
Deleted the function